File: week6/aaange/2467.js
`
- 분류 : 이분 탐색, 투 포인터
- 성능 요약
    - 메모리 : 26500 KB
    - 시간 : 200 ms
- 제출 일자 : 2025-02-24 17:28
`

const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const N = parseInt(input[0]);
const arr = input[1].split(" ").map(Number);

let left = 0;
let right = N - 1;
let min_value = Infinity;
let result = [];

while (left < right) {
    let mix = arr[left] + arr[right]
    if (Math.abs(mix) < min_value ) {
        min_value = Math.abs(mix)
        result = [arr[left], arr[right]]
    }

    if (mix > 0) {
        right--;
    } else {
        left++;
    }
}

console.log(result[0], result[1]);

File: week06/aaange/2467.py
"""
- 분류 : 이분 탐색, 투 포인터
- 성능 요약
    - 메모리 : 44748 KB
    - 시간 : 116 ms
- 제출 일자 : 2025-02-24 17:16
"""

N = int(input()) # 전체 용액의 수
arr = list(map(int, input().split()))

# 투 포인터 초기화
left = 0
right = N - 1
min_value = float("inf")
result = []

while left < right:
    mix = arr[left] + arr[right]

    # 현재 조합이 0에 더 가까우면 갱신
    if abs(mix) < min_value:
        min_value = abs(mix)
        result = [arr[left], arr[right]]

    if mix > 0:
        right -= 1
    else:
        left += 1
        

print(result[0], result[1])
